### PR TITLE
Add --includePubsubMessage option for PubSubToSplunk template.

### PR DESCRIPTION
**Pull request by melbrodrigues**
Optional parameter to forward a [PubSubMessage](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage) instead of just the underlying message data.

### CODE REVIEW

**Comment by melbrodrigues** [comment]
@googlebot I signed it!

File: src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java [hunk 1]
```diff
@@ -265,36 +291,66 @@ public void processElement(ProcessContext context) {
...
       return input
           .apply(
-              "ReadMessagesFromPubSub", PubsubIO.readStrings().fromSubscription(subscriptionName))
+              "ReadPubSubMessage",
+              PubsubIO.readMessagesWithAttributes().fromSubscription(subscriptionName))
+          .apply(
+              "ExtractMessageIfRequired",
+              ParDo.of(
+                  new DoFn<PubsubMessage, String>() {
+                    @Setup
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java:333]: Add a blank line.

File: src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java [hunk 2]
```diff
@@ -265,36 +291,66 @@ public void processElement(ProcessContext context) {
...
+                    public void setup() {
+                      if (includePubSubMessageValueProvider != null) {
+                        includePubSubMessage = includePubSubMessageValueProvider.get();
+                      }
+                      includePubSubMessage =
+                          MoreObjects.firstNonNull(
+                              includePubSubMessage, DEFAULT_INCLUDE_PUBSUBMESSAGE);
+                      LOG.info("Include PubSubMessage set to: {}", includePubSubMessage);
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java:341]: use the flag name here to make it clearer:

"includePubsubMessage set to {}"

File: src/main/java/com/google/cloud/teleport/templates/common/SplunkConverters.java [hunk 3]
```diff
@@ -111,6 +111,11 @@ public static FailsafeStringToSplunkEvent failsafeStringToSplunkEvent(
     ValueProvider<Integer> getParallelism();
 
     void setParallelism(ValueProvider<Integer> parallelism);
+  
+    @Description("Determines whether the template forwards a PubSubMessage or just the underlying data.")
+    ValueProvider<Boolean> getIncludePubSubMessage();
```
Comment (sabhyankar) [inline src/main/java/com/google/cloud/teleport/templates/common/SplunkConverters.java:116]: Can we correct the capitalization here?

getIncludePubsubMessage
setIncludePubsubMessage

i.e. We want to be consistent in referring to Pubsub without the upper case S.

File: src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java (hunk 1 above)
Comment (melbrodrigues) [reply src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java:333]: After `new DoFn<PubsubMessage, String>() {` right?

Comment (sabhyankar) [reply src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java:333]: Yes thats correct.

**Pull request merged**